### PR TITLE
Switch to bugsnag

### PR DESCRIPTION
**Pull request by harrison**

### CODE REVIEW

**Comment by harrison** [comment]
@jackfranklin reviews? Going update and tag you in all the clients after getting this in

File: ng-gc-base-app-service/bugsnag-config-service.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  function() {
+
+    this.config = function config(options, metaData) {
+      window.Bugsnag.releaseStage = options.releaseStage;
+      window.Bugsnag.apiKey = options.apiKey;
+      window.Bugsnag.notifyReleaseStages = options.notifyReleaseStages;
+      if (metaData) { window.Bugsnag.metaData = metaData; }
+    };
+
+    this.$get = angular.noop;
```
Comment (jackfranklin) [inline ng-gc-base-app-service/bugsnag-config-service.js]: what's the need for this?

File: ng-gc-base-app-service/bugsnag-config-service.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+  }
+]);
+
```
Comment (jackfranklin) [inline ng-gc-base-app-service/bugsnag-config-service.js]: :scissors:

File: ng-gc-base-app-service/exception-handler-config.js [hunk 3]
```diff
@@ -9,8 +9,8 @@ angular.module('ngGcExceptionHandlerProviderConfig', [
       '$delegate',
       function($delegate) {
         return function(exception, cause) {
-          window.Raven.captureException(exception);
           $delegate(exception, cause);
+          window.Bugsnag.notifyException(exception);
```
Comment (jackfranklin) [inline ng-gc-base-app-service/exception-handler-config.js]: `$window` for consistency?

File: ng-gc-base-app-service/ng-gc-base-app-service.spec.js [hunk 4]
```diff
@@ -21,21 +21,20 @@ describe('BaseAppService', function() {
...
       };
     });
-    beforeEach(module('ngGcRavenInitializer'));
+    beforeEach(module('ngGcBugsnagInitializer'));
     injectDependencies();
 
-    it('Raven.setUser', function() {
-      var user = {id: 1};
-      spyOn($window.Raven, 'setUser');
+    it('Bugsnag.user', function() {
```
Comment (jackfranklin) [inline ng-gc-base-app-service/ng-gc-base-app-service.spec.js]: would rather have a more descriptive `it`. `it('sets the bugsnag user correctly')` ?

**Comment by jackfranklin** [comment]
:+1: after small comments :)

File: ng-gc-base-app-service/bugsnag-config-service.js (hunk 1 above)
Comment (harrison) [inline ng-gc-base-app-service/bugsnag-config-service.js]: angular blows up if you don't have `$get` set

Comment (jackfranklin) [inline ng-gc-base-app-service/bugsnag-config-service.js]: :+1:

**Pull request merged**